### PR TITLE
DT-23155 Discharge Wizard Refactor and addition of tests

### PR DESCRIPTION
**Pull request by zacharymorel**
## Issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/23155
https://github.com/department-of-veterans-affairs/va.gov-team/issues/23156

## Description
Decision Tools recently took over the responsibility of maintaining the Discharge Wizard. 
This is a overhaul to add unit tests and refactor the nightwatch test to cypress. 
There has been some change of folder structure as well to bring it up to a current easier, navigable, directory structure. 

## Testing done
Ran app locally, ran cypress e2e, ran all unit tests

## Acceptance criteria
- [x] e2e tests pass
- [x] unit tests pass
- [x] App works as it does on production

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/discharge-wizard/constants/index.js [hunk 1]
```diff
@@ -100,48 +100,3 @@ export const prevApplicationYearCutoff = {
   3: 2011,
   4: 2017,
 };
-
-export const answerReview = (key, formValues) => {
```
Comment (zacharymorel) [inline src/applications/discharge-wizard/constants/index.js:104]: Note, this is moved to the helpers/index.js file and out of the constants.

File: src/applications/discharge-wizard/components/FormQuestions.jsx [hunk 2]
```diff
@@ -1,25 +1,23 @@
-import React from 'react';
+// Dependencies.
+import React, { Component } from 'react';
 import { range } from 'lodash';
 import PropTypes from 'prop-types';
 import { Link } from 'react-router';
 import Scroll from 'react-scroll';
 
+// Relative Imports
```
Comment (kelsonic) [inline src/applications/discharge-wizard/components/FormQuestions.jsx:8]: haha YES! My madness is spreading 😈 ❤️

File: src/applications/discharge-wizard/components/InstructionsPage.jsx [hunk 3]
```diff
@@ -1,8 +1,8 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { Link } from 'react-router';
 import recordEvent from 'platform/monitoring/record-event';
 
-class InstructionsPage extends React.Component {
+class InstructionsPage extends Component {
```
Comment (kelsonic) [inline src/applications/discharge-wizard/components/InstructionsPage.jsx:5]: :chef-kiss:

File: src/applications/discharge-wizard/constants/index.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import * as options from 'platform/static-data/options-for-select';
+export const DW_UPDATE_FIELD = 'DW_UPDATE_FIELD';
```
Comment (kelsonic) [inline src/applications/discharge-wizard/constants/index.js:1]: Can you scope this constant a bit more? Maybe something like:
```js
export const UPDATE_FIELD = 'discharge-wizard/UPDATE_FIELD'
```

That way if another app on VA.gov in the future also has an action that dispatches the constant `'UPDATE_FIELD'` they won't accidentally fire off the wrong action type 🙂

File: src/applications/discharge-wizard/reducers/discharge-wizard.js [hunk 5]
```diff
@@ -120,14 +120,14 @@ function nextQuestion(currentQuestion, answer, state) {
   return next;
 }
 
-function form(state = initialState, action) {
+export default (state = initialState, action) => {
```
Comment (kelsonic) [inline src/applications/discharge-wizard/reducers/discharge-wizard.js:123]: ❤️

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉 Lots of great work here and really well done! Only thing I mention in the comments is if you want, you can change `DW_UPDATE_FIELD` to `UPDATE_FIELD` as long as the string its value is assigned to is scoped a bit more (unique across all the apps we have), which can easily be done if you prefix something like `'discharge-wizard/UPDATE_FIELD'` or a variant thereof 🙂

File: src/applications/discharge-wizard/constants/index.js (hunk 4 above)
Comment (zacharymorel) [reply src/applications/discharge-wizard/constants/index.js:1]: Oh nice catch, I can add it.

**Pull request merged**